golang/pkg/dagent/utils/docker.go

```diff
@@ -608,3 +608,40 @@ func SecretList(ctx context.Context, prefix, name string) ([]string, error) {
 
 	return []string{}, nil
 }
+
+func ContainerCommand(ctx context.Context, command *common.ContainerCommandRequest) error {
+	operation := command.Operation
+	containerID := command.GetId()
+
+	cli, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
+	if err != nil {
+		return err
+	}
+
+	if operation == common.ContainerOperation_START_CONTAINER {
+		err = cli.ContainerStart(ctx, containerID, types.ContainerStartOptions{})
+	} else if operation == common.ContainerOperation_STOP_CONTAINER {
+		err = cli.ContainerStop(ctx, containerID, nil)
+	} else if operation == common.ContainerOperation_RESTART_CONTAINER {
+		err = cli.ContainerRestart(ctx, containerID, nil)
+	} else {
+		log.Printf("Unknown container operation %d on containerID %s", operation, containerID)
+	}
+
+	return err
+}
+
+func DeleteContainers(ctx context.Context, request *common.DeleteContainersRequest) error {
```

Comment:
Merging the two DeleteContainers function is not an option? Core will not call it without adding adding the container name, so no "prefix delete" will be initiated. Having two functions for almost the same thing looks weird.

Reply:
I didn't wanted to tweak with the core itself, cause i would had to modify the request. So my strat here is to get rid of the core and resolve this issue in the future.